File: 프로그래머스/2/42747. H－Index/H－Index.py
def solution(citations):
    answer = 0
    index = 0
    arr = sorted(citations, reverse=True)
    for i in range(len(arr)):
        if arr[i] < i+1:
            # H index 존재
            answer = i  # 인덱스를 넣어주잖아
            return answer
    return len(citations)

File: 프로그래머스/2/42747. H－Index/H－Index.java
import java.util.*;
class Solution {
    public int solution(int[] citations) {
        int index = 0;
        int size = citations.length;
        int answer = 0;
        Integer[] reverse = Arrays.stream(citations).boxed().toArray(Integer[]::new);
        
        Arrays.sort(reverse, (a, b) -> b - a); 
        for(int i = 0; i < size; i++) {
            if(reverse[i] < i+1) {
                answer = i;
                return answer;
            }
        }
        return size;
    }
}